### ISSUE TITLE
Running `pod lib lint` on Travis CI with `watchOS` target

### ISSUE THREAD

**Issue by cnoon** [CocoaPods/CocoaPods#4103]
Hey everyone,

I almost have Alamofire back up and running on Travis CI for our `swift-2.0` branch, but I'm having trouble with `pod lib lint` on our `watchOS` target and I'm not really sure how to fix it. I was hoping the experts could lend a hand. Here's all the different pieces that should be helpful in diagnosing:

### Podspec

```ruby
Pod::Spec.new do |s|
  s.name = 'Alamofire'
  s.version = '2.0.0-beta.3'
  s.license = 'MIT'
  s.summary = 'Elegant HTTP Networking in Swift'
  s.homepage = 'https://github.com/Alamofire/Alamofire'
  s.social_media_url = 'http://twitter.com/AlamofireSF'
  s.authors = { 'Alamofire Software Foundation' => 'info@alamofire.org' }
  s.source = { :git => 'https://github.com/Alamofire/Alamofire.git', :tag => s.version }

  s.ios.deployment_target = '8.0'
  s.osx.deployment_target = '10.9'
  s.watchos.deployment_target = '2.0'

  s.source_files = 'Source/*.swift'

  s.requires_arc = true
end
```

### Travis CI YAML File

```bash
language: objective-c
osx_image: xcode7
env:
  global:
  - LC_CTYPE=en_US.UTF-8
  - LANG=en_US.UTF-8
  matrix:
    - DESTINATION="OS=8.1,name=iPhone 4S" SCHEME="Alamofire iOS" SDK=iphonesimulator9.0 BUILD_EXAMPLE="YES" POD_LINT="YES"
    - DESTINATION="OS=8.2,name=iPhone 5" SCHEME="Alamofire iOS" SDK=iphonesimulator9.0 BUILD_EXAMPLE="YES" POD_LINT="NO"
    - DESTINATION="OS=8.3,name=iPhone 5S" SCHEME="Alamofire iOS" SDK=iphonesimulator9.0 BUILD_EXAMPLE="YES" POD_LINT="NO"
    - DESTINATION="OS=8.4,name=iPhone 6" SCHEME="Alamofire iOS" SDK=iphonesimulator9.0 BUILD_EXAMPLE="YES" POD_LINT="NO"
    - DESTINATION="OS=9.0,name=iPhone 6 Plus" SCHEME="Alamofire iOS" SDK=iphonesimulator9.0 BUILD_EXAMPLE="YES" POD_LINT="NO"
    - DESTINATION="arch=x86_64" SCHEME="Alamofire OSX" SDK=macosx10.11 BUILD_EXAMPLE="NO" POD_LINT="NO"
before_install:
  - gem install cocoapods --no-rdoc --no-ri --no-document --quiet
  - gem install xcpretty --no-rdoc --no-ri --no-document --quiet
script:
  - set -o pipefail
  - xcodebuild -version
  - xcodebuild -project Alamofire.xcodeproj -scheme "$SCHEME" -sdk "$SDK" -destination "$DESTINATION" 
    -configuration Debug ONLY_ACTIVE_ARCH=NO test | xcpretty -c
  - xcodebuild -project Alamofire.xcodeproj -scheme "$SCHEME" -sdk "$SDK" -destination "$DESTINATION" 
    -configuration Release ONLY_ACTIVE_ARCH=NO test | xcpretty -c
  - if [ $BUILD_EXAMPLE == "YES" ]; then 
      xcodebuild -project "iOS Example.xcodeproj" -scheme "iOS Example" -sdk "$SDK" -destination "$DESTINATION" 
      -configuration Debug ONLY_ACTIVE_ARCH=NO build | xcpretty -c; 
    fi
  - if [ $BUILD_EXAMPLE == "YES" ]; then 
      xcodebuild -project "iOS Example.xcodeproj" -scheme "iOS Example" -sdk "$SDK" -destination "$DESTINATION" 
      -configuration Release ONLY_ACTIVE_ARCH=NO build | xcpretty -c; 
    fi
  - if [ $POD_LINT == "YES" ]; then
      pod lib lint --verbose;
    fi
```

### Output from Travis CI when linting our watchOS target

```bash
-> Installing Alamofire (2.0.0-beta.3)
  - Running pre install hooks
Generating Pods project
  - Creating Pods project
  - Adding source files to Pods project
  - Adding frameworks to Pods project
  - Adding libraries to Pods project
  - Adding resources to Pods project
  - Linking headers
  - Installing targets
    - Installing target `Alamofire` watchOS 2.0
      - Generating Info.plist file at
      `../../../../../private/var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/Pods/Target
      Support Files/Alamofire/Info.plist`
      - Generating module map file at
      `../../../../../private/var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/Pods/Target
      Support Files/Alamofire/Alamofire.modulemap`
      - Generating umbrella header at
      `../../../../../private/var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/Pods/Target
      Support Files/Alamofire/Alamofire-umbrella.h`
    - Installing target `Pods` watchOS 2.0
      - Generating Info.plist file at
      `../../../../../private/var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/Pods/Target
      Support Files/Pods/Info.plist`
      - Generating module map file at
      `../../../../../private/var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/Pods/Target
      Support Files/Pods/Pods.modulemap`
      - Generating umbrella header at
      `../../../../../private/var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/Pods/Target
      Support Files/Pods/Pods-umbrella.h`
  - Running post install hooks
  - Writing Xcode project file to
  `../../../../../private/var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/Pods/Pods.xcodeproj`
  - Writing Lockfile in
  `../../../../../var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/Podfile.lock`
  - Writing Manifest in
  `../../../../../private/var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/Pods/Manifest.lock`
  - Running post install hooks
   Building with xcodebuild. 
xcodebuild clean build -target Pods 2>&1
=== CLEAN TARGET Alamofire OF PROJECT Pods WITH THE DEFAULT CONFIGURATION (Release) ===
Check dependencies
[BEROR]Code Sign error: No code signing identities found: No valid signing identities (i.e. certificate and private key pair) were found.
Create product structure
/bin/mkdir -p /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Alamofire.framework/Modules
/bin/mkdir -p /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Alamofire.framework/Headers
Clean.Remove clean /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Alamofire.framework.dSYM
    builtin-rm -rf /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Alamofire.framework.dSYM
Clean.Remove clean /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Pods.build/Release-watchos/Alamofire.build
    builtin-rm -rf /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Pods.build/Release-watchos/Alamofire.build
Clean.Remove clean /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Alamofire.framework
    builtin-rm -rf /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Alamofire.framework
=== CLEAN TARGET Pods OF PROJECT Pods WITH THE DEFAULT CONFIGURATION (Release) ===
Check dependencies
[BEROR]Code Sign error: No code signing identities found: No valid signing identities (i.e. certificate and private key pair) were found.
Create product structure
/bin/mkdir -p /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Pods.framework
/bin/mkdir -p /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Pods.framework/Headers
Clean.Remove clean /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Pods.framework
    builtin-rm -rf /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Pods.framework
Clean.Remove clean /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Pods.framework.dSYM
    builtin-rm -rf /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Release-watchos/Pods.framework.dSYM
Clean.Remove clean /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Pods.build/Release-watchos/Pods.build
    builtin-rm -rf /var/folders/vy/rcv48w3j4w79llzf_x6qnvw40000gn/T/CocoaPods/Lint/build/Pods.build/Release-watchos/Pods.build
** CLEAN SUCCEEDED **
=== BUILD TARGET Alamofire OF PROJECT Pods WITH THE DEFAULT CONFIGURATION (Release) ===
Check dependencies
Code Sign error: No code signing identities found: No valid signing identities (i.e. certificate and private key pair) were found.
** BUILD FAILED **
The following build commands failed:
	Check dependencies
(1 failure)
 -> Alamofire (2.0.0-beta.3)
    - ERROR | [watchOS] Returned an unsuccessful exit code.
    - NOTE  |  [BEROR]Code Sign error: No code signing identities found: No valid signing identities (i.e. certificate and private key pair) were found.
    - NOTE  | [watchOS]  Code Sign error: No code signing identities found: No valid signing identities (i.e. certificate and private key pair) were found.
[!] Alamofire did not pass validation, due to 1 error.
You can use the `--no-clean` option to inspect any issue.
```

> Here's the [link](https://travis-ci.org/Alamofire/Alamofire/jobs/77620322) to the full build log.

### Screenshot of watchOS code signing identities

![screen shot 2015-08-29 at 9 55 48 am](https://cloud.githubusercontent.com/assets/169110/9563309/2deabb68-4e34-11e5-83a7-2de26613e083.png)

### Summary

I'm not really sure what else to provide. I'm able to successfully lint locally since I have a valid code signing identity, but I cannot lint properly on Travis CI. I've been digging around your issue history and have found that using `pod lib lint --quick` will solve the issue. I have been able to verify that `pod lib lint --quick` does in fact fix my code signing issue. However, I feel that that is only a temporary workaround. 

Any help would be much appreciated as always!

**Comment by segiddins** [comment]
https://github.com/CocoaPods/CocoaPods/blob/0c9f24807f53a93bff0628f799ade47a1a4f7f1c/lib/cocoapods/validator.rb#L612 just needs to be changed to also include `watchos`

**Comment by vittoriom** [comment]
I have the same problem here, trying to push a podspec to trunk but can't code sign the watchos part?

**Comment by kylef** [comment]
This has been resolved by #4130.
